tools/build/Makefile.vars

```diff
@@ -286,7 +286,9 @@ COMMON_CFLAGS += -g -fno-lto
 endif
 
 ifeq (y,$(BUILD_TYPE_RELEASE))
-COMMON_CFLAGS += $(RELEASE_COMMON_CFLAGS)
+ifeq (,$(shell echo $(COMMON_CFLAGS) | grep -e "-O0"))
+COMMON_CFLAGS += $(FORTIFY_CFLAGS)
+endif
```

Comment:
Additional info: The use case for this is the situation where the user starts the build process with
```npm install``` or ```npm install --debug```. In both cases the npm install process ends up calling ```make alldefconfig``` followed by ```make```, but prefixed with a bunch of configuration options. In case of ```npm install```, it results in

```bash
USE_NODEJS="y" RPATH="y" make
```

whereas in the ```npm install --debug``` case it also has to at least turn on ```-g -O0``` in Soletta. It does this by also adding ```COMMON_CFLAGS="-g -O0"``` to the settings preceding the ```make```.

I also don't necessarily want to link Soletta's debug build feature to node.js' debug build, considering ```-g -O0``` as sufficient debugging support by Soletta for the node.js bindings.

Here's how it is currently: https://github.com/solettaproject/soletta/blob/5daa00eadbe275dbb696a749f22fd319e39d877f/bindings/nodejs/build-for-npm.sh#L28-L34

There are two problems with how it is currently:
0. It modifies the .config file. This is fixed in this PR by conditionally setting COMMON_CFLAGS for ```make``` instead.
0. ```-O0``` from external COMMON_CFLAGS is being overridden by RELEASE_COMMON_CFLAGS which later adds -O2 so as to support -DFORTIFY_SOURCE=2. This currently cannot be avoided by either modifying .config or by prefixing ```make``` with a variable AFAICT.